File: Sortings/selectionSort.java
package Sortings;

public class selectionSort {
    

    public static void selectionSort(int[] arr){
        int n = arr.length;
        for(int i = 0; i < n - 1; i++){
            int minIndex = 1;
            for(int j = i + 1; j < n; j++){
                if(arr[j] < arr[minIndex]){
                    minIndex = j;
                }
            }
            // Swap the found minimum element with the first element of the unsorted array
            int temp = arr[minIndex];
            arr[minIndex] = arr[i];
            arr[i] = temp;
        }
        System.out.println("After selection sort:");
        for (int i = 0; i < n; i++) {
            System.out.print(arr[i] + " ");
        }
    }
    public static void main(String[] abc){
        int[] arr = {13, 46, 24, 52, 20, 9};
        int n = arr.length;
        System.out.println("Before selection sort:");
        for (int i = 0; i < n; i++) {
            System.out.print(arr[i] + " ");
        }
        System.out.println();
        selectionSort(arr);

    }
}

File: Sortings/selectionSort.py
def selectionSort(arr):
    n = len(arr)
    for i in range(n - 1):
        min_index = i
        for j in range(i + 1, n):
            if arr[j] < arr[min_index]:
                min_index = j
                
        # Swapping the found minimum element with the first element of the unsorted array
        temp = arr[min_index]
        arr[min_index] = arr[i]
        arr[i] = temp
        
    print("Sorted array is:")
    for i in range(n):
        print(arr[i], end=" ")
    
# Example usage
if __name__ == "__main__":
    arr = [64, 25, 12, 22, 11]
    selectionSort(arr)